TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/AssetController.php

```diff
@@ -185,30 +192,74 @@ public function relatedNodesAction(AssetInterface $asset)
 
         $usageReferences = $this->assetService->getUsageReferences($asset);
         $relatedNodes = [];
+        $inaccessibleRelations = [];
 
-        /** @var AssetUsageInNodeProperties $usage */
         foreach ($usageReferences as $usage) {
-            $documentNodeIdentifier = $usage->getDocumentNode() instanceof NodeInterface ? $usage->getDocumentNode()->getIdentifier() : null;
-
-            $relatedNodes[$usage->getSite()->getNodeName()]['site'] = $usage->getSite();
-            $relatedNodes[$usage->getSite()->getNodeName()]['documentNodes'][$documentNodeIdentifier]['node'] = $usage->getDocumentNode();
-            $relatedNodes[$usage->getSite()->getNodeName()]['documentNodes'][$documentNodeIdentifier]['nodes'][] = [
-                'node' => $usage->getNode(),
-                'nodeData' => $usage->getNode()->getNodeData(),
-                'contextDocumentNode' => $usage->getDocumentNode(),
-                'accessible' => $usage->isAccessible()
+            $inaccessibleRelation = [
+                'type' => get_class($usage),
+                'usage' => $usage
+            ];
+
+            if (!$usage instanceof AssetUsageInNodeProperties) {
+                $inaccessibleRelations[] = $inaccessibleRelation;
+                continue;
+            }
+
+            $nodeType = $this->nodeTypeManager->getNodeType($usage->getNodeTypeName());
+            $workspace = $this->workspaceRepository->findByIdentifier($usage->getWorkspaceName());
+            $accessible = $this->domainUserService->currentUserCanReadWorkspace($workspace);
+
+            $inaccessibleRelation['nodeIdentifier'] = $usage->getNodeIdentifier();
+            $inaccessibleRelation['workspaceName'] = $usage->getWorkspaceName();
+            $inaccessibleRelation['workspace'] = $workspace;
+            $inaccessibleRelation['nodeType'] = $nodeType;
+            $inaccessibleRelation['accessible'] = $accessible;
+
+            if (!$accessible) {
+                $inaccessibleRelations[] = $inaccessibleRelation;
+                continue;
+            }
+
+            $node = $this->getNodeFrom($usage);
+            // this should actually never happen.
+            if (!$node) {
+                $inaccessibleRelations[] = $inaccessibleRelation;
+                continue;
+            }
+
+            $site = $node->getContext()->getCurrentSite();
```

Comment:
This always return the main site so references from different sites will be in the same group in the output.
At least this is valid for 2.3.* and maybe some 3.* versions.
Would be great if this is either fixed for older versions or we don't show false information at least.

Reply:
I addressed this in a follow-up commit to this pull request.